src/search_renderer.cpp

```diff
@@ -93,37 +93,42 @@ std::string SearchRenderer::getHtml()
   kainjow::mustache::data pages{kainjow::mustache::data::type::list};
 
   auto resultStart = mp_searcher->getResultStart();
-  auto resultEnd = mp_searcher->getResultEnd();
-  auto resultCountPerPage = resultEnd - resultStart;
+  auto pageLength = this->pageLength;
   auto estimatedResultCount = mp_searcher->getEstimatedResultCount();
   auto currentPage = 0U;
   auto pageStart = 0U;
   auto pageEnd = 0U;
   auto lastPageStart = 0U;
-  if (resultCountPerPage) {
-    currentPage = resultStart/resultCountPerPage;
+  if (pageLength) {
+    currentPage = resultStart/pageLength;
     pageStart = currentPage > 4 ? currentPage-4 : 0;
     pageEnd = currentPage + 5;
-    if (pageEnd > estimatedResultCount / resultCountPerPage) {
-      pageEnd = estimatedResultCount / resultCountPerPage;
+    if (pageEnd > estimatedResultCount / pageLength) {
+      pageEnd = estimatedResultCount / pageLength;
     }
-    if (estimatedResultCount > resultCountPerPage) {
-      lastPageStart = static_cast<int>(round(estimatedResultCount/resultCountPerPage)) * resultCountPerPage;
+    if (estimatedResultCount > pageLength) {
+      lastPageStart = static_cast<int>(round(estimatedResultCount/pageLength)) * pageLength;
     }
+    if(pageEnd*pageLength==estimatedResultCount && pageEnd>0)pageEnd--;
   }
 
-  for (unsigned int i = pageStart; i < pageEnd; i++) {
+  for (unsigned int i = pageStart; i <= pageEnd; i++) {
```

Comment:
Did this change stay after attempts to find a working solution?

Reply:
yes as now I am not adding last page individually it should be added using the loop only

Reply:
@MananJethwani I cannot agree here. If `pageEnd` stays equal to `currentPage+5` then this change includes an extra page in the page list. I think that the problem you were dealing with was about the case when `pageEnd` had to be adjusted, so instead it just needs to be adjusted properly.